File: src/script/project_euler/55.cpp
#include <iostream>
/*There is an overflow problem here that needs to be solved!! */
int reverse(int b) {
	int r = 0;
	while(b != 0) {
		r = 10 * r + (b % 10);
		b /= 10;
	}
	return r;
}

int isPalindrome(int c) {
	if(c == reverse(c)) return 1;
	else return 0;
}

int isLychrel(int a) {
	int reva, i, c;
	int b = a;
	for(i = 0;i < 50;i ++) {
		c = b + reverse(b);
		if(isPalindrome(c)) return 0;
		b = c;
	}
	return 1;
}
int main() {
	int i, count = 0;
	
	for(i = 1;i < 10000;i ++) {
		if(isLychrel(i)) count ++;
	}
	cout << count << endl;
	return 0;
}

File: src/script/project_euler/55.py
def rev(n):
    rev_n = 0
    while n != 0:
        units_place_digit = n % 10
        rev_n = rev_n * 10 + units_place_digit
        n = n / 10
    return rev_n

def ispalindrome(n):
    if rev(n) == n:
        return True
    else:
        return False

def islychrel(n):
    i = 0
    while i < 50:
        b = n + rev(n)
        if ispalindrome(b):
            return False
        n = b
        i = i + 1
    return True

def main():
    count = 0
    for n in range(1, 10000):
        if islychrel(n):
            count = count + 1
    print count

if __name__ == '__main__':
    main()
